cmd/e2e-test/basic_test.go

```
@@ -239,3 +239,108 @@ func whiteboxTest(ing *v1beta1.Ingress, s *e2e.Sandbox, t *testing.T, region str
+				t.Fatalf("e2e.NewCert(%q, %q, _, %t) = %v, want nil", certName, host, false, err)
+			}
+			if err := cert.Create(s); err != nil {
+				t.Fatalf("cert.Create(_) = %v, want nil, error creating cert %s", err, cert.Name)
+			}
+			t.Logf("Cert created %s", certName)
+			defer cert.Delete(s)
+
+			ing := fuzz.NewIngressBuilder(s.Namespace, "ing1", "").
+				AddPath(host, "/", svcName, port80).AddTLS([]string{}, cert.Name).Build()
+
+			crud := e2e.IngressCRUD{C: Framework.Clientset}
+			if _, err := crud.Create(ing); err != nil {
+				t.Fatalf("crud.Create(%s/%s) = %v, want nil; Ingress: %v", ing.Namespace, ing.Name, err, ing)
+			}
+			t.Logf("Ingress created (%s/%s)", s.Namespace, ing.Name)
+			ing = waitForStableIngress(true, ing, s, t)
+			gclb := whiteboxTest(ing, s, t, "")
+			vip := ing.Status.LoadBalancer.Ingress[0].IP
+
+			// Update ingress with desired frontend resource configuration.
+			ingBuilder := fuzz.NewIngressBuilderFromExisting(ing)
+			if tc.disableHTTP {
+				ingBuilder = ingBuilder.SetAllowHttp(false)
+			}
+			if tc.disableHTTPS {
+				ingBuilder = ingBuilder.SetTLS(nil)
+			}
+			ing = ingBuilder.Build()
+
+			if _, err := crud.Update(ing); err != nil {
+				t.Fatalf("update(%s/%s) = %v, want nil; ingress: %v", ing.Namespace, ing.Name, err, ing)
+			}
+			t.Logf("Ingress updated (%s/%s)", ing.Namespace, ing.Name)
+			ing = waitForStableIngress(true, ing, s, t)
+
+			deleteOptions := &fuzz.GCLBDeleteOptions{
+				SkipDefaultBackend:          true,
+				CheckHttpFrontendResources:  tc.disableHTTP,
+				CheckHttpsFrontendResources: tc.disableHTTPS,
+			}
+			// Wait for unused frontend resources to be deleted.
+			if err := e2e.WaitForFrontendResourceDeletion(ctx, Framework.Cloud, gclb, deleteOptions); err != nil {
+				t.Errorf("e2e.WaitForIngressDeletion(..., %q, _) = %v, want nil", ing.Name, err)
+			}
+
+			// Special case in which we whitebox test just the number of forwarding rules.
+			if tc.disableHTTP && tc.disableHTTPS {
```

This is sort of out of scope of this PR but I'm worried this is not the right behavior. If someone tries to turn off both HTTP & HTTPS, I feel like we should return an error to them rather than garbage collecting both of their forwarding rules.

---

This may require changes to existing behavior, currently we emit a warning:
https://github.com/kubernetes/ingress-gce/blob/a2e96e0dbedf9c5929c0202e1c8a7fd4ef92deb4/pkg/loadbalancers/l7.go#L200-L202
The fix would return a warning too when user disables both HTTP and HTTPS.

We either switch to a workflow where we return an error on both creation and update or we keep the existing workflow. 

Is it worth changing the existing workflow to return an error if both HTTP & HTTPS are disabled?.

---

Let me rephrase what I think the behavior should be. In my mind, disabling both HTTP & HTTPS is a mistake that the user makes since its not valid config. If a user truly wanted both frontends to be deleted when this happens, they should just delete the Ingress outright.

Disabling both protocols ideally should be an error than is propagated all the way up. I don't see any issues with this but please keep me honest.

---

Raised a PR to fix this: https://github.com/kubernetes/ingress-gce/pull/950

The only concern is that we re-queue the ingress repeatedly until user fixes the configuration.

---

I think that's fine. This is no different than another situation w/ invalid config such as referencing a service that does not exist.

---

The fix is merged now. Updated this test to exclude the test case.